apps/studio/src/components/TabQueryEditor.vue

```diff
@@ -746,8 +748,32 @@
             }
           })
 
+
+
+
           if (this.userKeymap === "vim") {
             const codeMirrorVimInstance = document.querySelector(".CodeMirror").CodeMirror.constructor.Vim
+            const coreTabs = this.$root.$refs.CoreTabs
```

Comment:
This was pretty clever, but we have a central event bus that we use to communicate across components for tasks like these: `this.$root.$emit(AppEvent.newTab)`. You can see we use this in many different places.